index.html

```diff
@@ -1691,7 +1691,72 @@ <h3>Add Proof</h3>
         </p>
 
       </section>
+      <section>
+        <h3>Add Proof Set</h3>
+        <p>
+          The following algorithm specifies how to add a proof set to a 
+          document. Required inputs are an
+          <a>unsecured data document</a> (<var>unsecuredDocument</var>) and a
+          (unordered) list of
+          <a>proof options</a> (<var>options_list</var>). Each <a>proof options</a>
+          (<var>options</var>) in the list must satisfy the criteria in 
+          section <a href="#add-proof"></a>.
+          A <dfn>secured data document</dfn> is produced as
+          output. Whenever this algorithm encodes strings, it MUST use UTF-8 encoding.
+        </p>
 
+        <ol class="algorithm">
+          <li>
+Let <var>output</var> be a copy of <var>unsecuredDocument</var> and 
+<var>proof_list</var> be an empty list.
+          </li>
+          <li>
+For each <var>options</var> in <var>options_list</var> run steps 2 through 8 of
+the algorithm in section <a href="#add-proof"></a>. If no exceptions are raised 
+append the generated <var>proof</var> value to the <var>proof_list</var> 
+otherwise raise the exception.
+          </li>
+          <li>
+Set <var>output</var>.<var>proof</var> to the value of <var>proof_list</var>.
+          </li>
+          <li>
+Return <var>output</var> as the <a>secured data document</a>.
+          </li>
+        </ol>
+      </section>
+      <section>
+        <h3>Add Proof Chain</h3>
+        <p>
+          The following algorithm specifies how to add a proof chain to a 
+          document. Required inputs are an
+          <a>unsecured data document</a> (<var>unsecuredDocument</var>) and a
+          (unordered) list of
+          <a>proof options</a> (<var>options_list</var>) to which an order will
+          be established. Each <a>proof options</a>
+          (<var>options</var>) in the list must satisfy the criteria in 
```

Comment:
The `must` in the last line here should probably be changed, either to `MUST` or to a non-normative word.
```suggestion
          list (unordered) of
          <a>proof options</a> (<var>options_list</var>) over which an order will
          be established. Each <a>proof option</a>
          (<var>option</var>) in the list must satisfy the criteria in 
```